draft-pda-core.md

```diff
@@ -291,11 +291,437 @@ Each PA task in this document is divided into three sub-protocols as follows.
+`i`-th share for each `0 < i <= L`, where `L` is the length of
+`PAVerifyStartReq.payloads`.
+
+[TODO: Instead of a sequence of shares and a sequence of payloads,
+`PAVerifyStartReq` should have a sequence of (share, payload) pairs. (Here and
+below.) This is difficult to express in TLS syntax because we have to select on
+the protocol type.]
+
+**Response.**
+The helper handles well-formed requests as follows. (As usual, malformed
+requests are handled as described in {{pa-error-common-aborts}}.) It first looks
+for the PA parameters `PAParam` for which `PAVerifyStartReq.task.id ==
+PAParam.task.id`. Next, it looks up the HPKE config and corresponding secret key
+associated with `PAVerifyStartReq.key_config_id`. If not found, then it aborts
+and alerts the leader with "unrecognized key config".
+
+Finally, for each pair of shares and message payloads, the helper does as
+follows. Let `share` denote the share. It computes the HPKE context as
+
+~~~
+context = SetupBaseR(share.enc, sk, PAParam.task)
+~~~
+
+where `sk` is the secret key corresponding to the HPKE key config. For each
+report share `share`, it derives is share of the input and proof from `context`
+and `share.payload` and computes its response according to the PA protocol. It
+responds to the POST request from the leader with status 200 and the following
+message:
+
+~~~
+struct {
+  PAProto proto;
+  select (PAVerifyStartResp.proto) {
+    case prio: PrioVerifyStartResp payloads<0..2^24-1>;
+    case hits: HitsVerifyStartResp payloads<0..2^24-1>;
+  }
+} PAVerifyStartResp;
+~~~
+
+The `i`-th element of `PAVerifyStartResp.payloads` corresponds to the `i`-th element
+of `PAVerifyStartReq.payloads` for each `0 < i <= L`.
+
+### Verify Finish
+
+Next, the leader processes each message in `PAVerifyStartResp.payloads`
+according to the PA protocol and sends a request to the helper, constructed as
+follows.
+
+**Request.**
+The leader sends a POST request to `[helper]/verify_finish` with the following
+message:
+
+~~~
+struct {
+  PATask task;                     // Equal to PAVerifyStartReq.task
+  uint8 key_config_id;             // Equal to PAVerifyStartReq.key_config_id
+  PAHelperShare shares<0..2^24-1>; // Equal to PAVerifyStartReq.shares
+  select (PAVerifyFinishReq.task.proto) {
+    case prio: PrioVerifyFinishReq payloads<0..2^24-1>;
+    case hits: HitsVerifyFinishReq payloads<0..2^24-1>;
+  }
+} PAVerifyFinishReq;
+~~~
+
+As usual, the `i`-th payload corresponds to the `i`-th payload of the
+`PAVerifyStartResp` sent by the helper in response to the previous request.
+
+**Response.**
+The helper handles POST requests to `[helper]/verify_finish` as follows. It
+begins just as before by looking up the PA parameters `PAParam` for which
+`PAVerifyStartReq.task.id == PAParam.task.id`. Next, it looks up the HPKE config
+and corresponding secret key associated with `PAVerifyStartReq.key_config_id`.
+If not found, then it aborts and alerts the leader with "unrecognized key
+config".
```

Comment:
Is there missing text here? What does the helper actually do with the FinishReq message?